tools/java/org/unicode/cldr/util/XPathParts.java

```diff
@@ -1049,16 +1070,17 @@ public XPathParts set(XPathParts parts) {
         if (frozen) {
             throw new UnsupportedOperationException("Can't modify frozen Element");
         }
-        try {
-            dtdData = parts.dtdData;
-            elements.clear();
-            for (Element element : parts.elements) {
-                elements.add((Element) element.clone());
-            }
-            return this;
-        } catch (CloneNotSupportedException e) {
-            throw (InternalError) new InternalError().initCause(e);
+
+        dtdData = parts.dtdData;
+        elements.clear();
```

Comment:
Note that an unfrozen XPathParts might have an immutable list — so you have to do:

if (!elements.isEmpty()) {
  makeMutable();
  elements.clear();
}